course6/toyapp_thomasdhuys/drawlib/vector.h

```diff
@@ -0,0 +1,51 @@
+#pragma once
+
+#include <cassert>
+#include <iostream>
+#include <array>
+
+namespace Draw {
+
+template<size_t X, typename T>
+class VectorXD {
+public:
+    VectorXD() {}
+    VectorXD(VectorXD<X-1,T> v, T e)
+    {
+       std::copy(v.begin(), v.end(), this->begin());
+       data[X-1] = e;
+    }
+    VectorXD(std::initializer_list<T> il)
+    {
+        //assert(args.size() == X); //, "Incorrect number of arguments");
+        std::copy(il.begin(), il.end(), data.begin());
+    }
+
+	T& operator[](std::size_t index) {return data.at(index);}
+	const T& operator[](std::size_t index) const {return data.at(index);}
+
+    auto begin() {return data.begin();}
+    auto end()   {return data.end();}
+
+protected:
+    std::array<T, X> data;
+};
```

Comment:
Good, but I would expect the vector to have some useful operators, like +, -, == etc.